### PR TITLE
(#3446) Update and Restore Grants & Contracts Chart

### LINKED ISSUE #3446: Update and Restore Highchart: Grant & Contract Awards / Institutions (FY22)
SPs: 5

#### Summary
__Issue:__ The chart on the "[Grant and Contract Awards](https://www.cancer.gov/about-nci/budget/fact-book/extramural-programs/grant-contract-awards)" page under the "Grant and Contract Awards by State and Institution" section should be replaced by a highchart that contains the correct data. The data initially included in the highchart under the "Grant and Contract Awards by State and Institution" section contained inaccurate data. As a non-dev, short-term solution, we replaced the highchart with a static table that contains the correct data. 

- New (correct) data that should be reflected in the chart: [awards-by-state-and-institution-fy21 (updated 2022.06.15).xlsx](https://github.com/NCIOCPL/cgov-digital-platform/files/9939832/awards-by-state-and-institution-fy21.updated.2022.06.15.xlsx)
- Old (incorrect) data that should not be reflected in the chart: [OLD awards-by-state-and-institution-fy21 (retired 2022.06.15).xlsx](https://github.com/NCIOCPL/cgov-digital-platform/files/9939869/OLD.awards-by-state-and-institution-fy21.retired.2022.06.15.xlsx)

__Goal:__ To restore the highchart on the "[Grant and Contract Awards](https://www.cancer.gov/about-nci/budget/fact-book/extramural-programs/grant-contract-awards)" page under the "Grant and Contract Awards by State and Institution" section that contains accurate data (found in the attached Excel document). 

**Parent Issue:** N/A

**Tasks**
- [x] Pre-requisite: #3761 released to production
- [x] Reinstate high chart on the "[Grant and Contract Awards](https://www.cancer.gov/about-nci/budget/fact-book/extramural-programs/grant-contract-awards)" page in place of static table
- [x] Correct bug originally reported in #3329
- [x] Implement data for FY2022 
- [x] Verify data is correct and that the chart loads properly

**Note**
- When testing high chart, add test case to ensure issue reported in #3329 is addressed. 
- Initial Jira ticket for this request found here: https://tracker.nci.nih.gov/browse/CGOV-16326

### What's the updated acceptance criteria?
```gherkin
Scenario: Display Grant & Contracts Awards/Institutions HighChart

Given a user is viewing the page with the Grant & Contracts Awards/Institutions chart
When the user views the page
Then the chart loads
  And the chart loads every time!
```

### PR DESCRIPTION
**Pull request by seyilonge-nci**
* Fixed bug originally reported on #3329 (Added loading of data plugin and json file for geo location)
* Added data to Highcharts map object
* Updated json data file for FY 2022 and yaml content file

Closes #3446

### CODE REVIEW

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/library/bypass-appropriations.js [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-    series: appropriationsData.series,
-    tooltip: appropriationsData.tooltip
+    chart: data.chartType,
+    title: data.chartTitle,
+    subtitle: data.chartSubTitle,
+    xAxis: data.xAxis,
+    plotOptions: data.plotOptions,
+    yAxis: data.yAxis,
+    series: data.series,
+    tooltip: data.tooltip
```
Comment (blairlearn) [inline docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/library/bypass-appropriations.js:15]: This change is unrelated to the bug fix.  Please undo it.

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/index.js [hunk 2]
```diff
@@ -48,11 +48,19 @@ const getChartData = async (dataFileName, chartRevision) => {
...
-    const response = await axios.get(fileURL);
-    const { data } = response;
-    return data;
+    const requests = requestURLArray.map(requestURL => axios.get(requestURL));
+
+    const [chartConfigResponse, geoDataResponse] = await axios.all(requests);
+    const data = chartConfigResponse?.data;
+    const miscData = geoDataResponse?.data;
+    return {data, miscData};
   } catch (error) {
```
Comment (blairlearn) [inline docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/index.js:64]: The naming for the supplemental data gets confusing.  It starts off as a "miscellaneous data URL".  Then it's retrieved as "geo data", then it's back to being stored as "miscellaneous."

Calling it "geo data" is wrong; that just reflects the specific bug you're working on right now. I'm also distrustful of the "miscellaneous" name, though it's definitely more appropriate than "geo data".

**Review by blairlearn — CHANGES_REQUESTED** [review]
This change goes beyond the scope of a bug fix and refactors every chart. It's not clear why that's necessary.

Every chart appears to go from this
```javascript
function initChart(Chart, htmlAmountData) {
```

to this
```javascript
function initChart(Chart, chartData) {
  const { data } = chartData;
```

If all the charts are extracting the data object, wouldn't it make more sense to pass only the data object?

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/index.js (hunk 2 above)
Comment (seyilonge-nci) [reply docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/index.js:64]: This I actually intended renaming. Missed out. Will fix.

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/library/bypass-appropriations.js (hunk 1 above)
Comment (blairlearn) [reply docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/library/bypass-appropriations.js:15]: I'll partially retract that.  This is actually part of the overall pattern of passing extra data and refactoring all the charts to accommodate it.

**Comment by seyilonge-nci** [comment]
> This change goes beyond the scope of a bug fix and refactors every chart. It's not clear why that's necessary.
> 
> Every chart appears to go from this
> 
> ```js
> function initChart(Chart, htmlAmountData) {
> ```
> 
> to this
> 
> ```js
> function initChart(Chart, chartData) {
>   const { data } = chartData;
> ```
> 
> If all the charts are extracting the data object, wouldn't it make more sense to pass only the data object?

The bugfix called for a refactor which touched code that affects the arguments for `initChart`. `grants-contracts` chart extracts not just the `data` object alone, it also gets the `miscData` object for the map.

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/library/bypass-appropriations.js (hunk 1 above)
Comment (seyilonge-nci) [reply docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/library/bypass-appropriations.js:15]: This goes to the same issue you mentioned above which I addressed in the earlier reply.

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/index.js (hunk 2 above)
Comment (seyilonge-nci) [reply docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/index.js:64]: @blairlearn Fixed!

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/library/bypass-appropriations.js (hunk 1 above)
Comment (seyilonge-nci) [reply docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/library/bypass-appropriations.js]: @blairlearn Reverted changes made to the other chart files. Now just passing in new argument to function for `grants-contracts` chart.

**Comment by alinai** [comment]
Looks good on ODE702.  Thanks @seyilonge-nci 
Per @seyilonge-nci we should ignore the "Cannot destructure property..." error as that applies to all charts, and comes from Highcharts itself. 
cc: @santwuma , @KateMashkinaNIH

**Review by blairlearn — APPROVED** [review]
